common/txmgr/types/chain.go

```diff
@@ -4,7 +4,10 @@ import "fmt"
 
 // Sequence represents the base type, for any chain's sequence object.
 // It should be convertible to a string
-type Sequence fmt.Stringer
+type Sequence interface {
+	fmt.Stringer
+	Int64() int64
```

Comment:
Not the bounds - the type itself. This imposes [linear/total order](https://en.wikipedia.org/wiki/Total_order) (even more actually) - is that always mappable?
With only `fmt.Stringer`, we were essentially only asking for a kind of `comparability`, and a linear/total order only requires `constraints.Ordered`, but integer goes even further:
```mermaid
flowchart
subgraph integer
subgraph constraints.Ordered
comparable
end
end
```
How would a UTXO model work?

Reply:
From another perspective, if this reduction to integer is valid for all cases, why did we generalize away from that in the first place? I think I'm missing something :thinking:

Reply:
ah that's a good point - I think i'll need to rethink having the `Int64()` method since you're right in pointing out that the UTXO model would not fit this

Reply:
I'm going to make a note in this ticket to re-evaluate this: https://smartcontract-it.atlassian.net/browse/BCI-860

we want confirmation logic to be able to support more than just sequential nonces - so it will probably come into play with this ticket

Reply:
Yes, the Int64() is only temporary, to allow nonce comparison operator currently.
We need to either introduce a comparator for Sequence, or do away with the logic in TXM, that requires a nonce comparison.